src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -2491,15 +2542,38 @@ object Desugar {
 
     }
 
-    def permissionD(ctx: FunctionContext)(perm: PPermission): Writer[in.Permission] = {
-      val src: Meta = meta(perm)
+    def permissionD(ctx: FunctionContext)(exp: PExpression): Writer[in.Expr] = {
+      maybePermissionD(ctx)(exp) getOrElse exprD(ctx)(exp)
+    }
+
+    def maybePermissionD(ctx: FunctionContext)(exp: PExpression): Option[Writer[in.Expr]] = {
+      val src: Meta = meta(exp)
       def goE(e: PExpression): Writer[in.Expr] = exprD(ctx)(e)
 
-      perm match {
-        case PFullPerm() => unit(in.FullPerm(src))
-        case PNoPerm() => unit(in.NoPerm(src))
-        case PFractionalPerm(left, right) => for { l <- goE(left); r <- goE(right) } yield in.FractionalPerm(l, r)(src)
-        case PWildcardPerm() => unit(in.WildcardPerm(src))
+      exp match {
+        case n: PInvoke => Some(info.resolve(n) match {
+          case Some(_: ap.Conversion) =>
+            for {
+              // the well-definedness checker ensures that there is exactly one argument
+              arg <- permissionD(ctx)(n.args.head)
+            } yield in.Conversion(in.PermissionT(Addressability.conversionResult), arg)(src)
+          // this case cannot be None to avoid cycles when desugaring function calls that return permissions
+          case _ => invokeD(ctx)(exp)
+        })
```

Comment:
why is this case needed? Is this not just covered by exprD?

Reply:
this is the leftover of a previous try to remove the ignorePerm argument, before I moved the `if` statement at the beginning of `exprD`. I removed it in the latest commit